erpnext/hr/report/bank_remittance/bank_remittance.py

```diff
@@ -0,0 +1,168 @@
+
+	for salary in salary_slips:
+		if salary.bank_name and salary.bank_account_no and salary.debit_acc_no and salary.status in ["Submitted", "Paid"]:
+			row = {
+					"payroll_no": salary.payroll_entry,
+					"debit_account": salary.debit_acc_no,
+					"payment_date": frappe.utils.formatdate(salary.modified.strftime('%Y-%m-%d')),
+					"bank_name":salary.bank_name,
+					"employee_account_no":salary.bank_account_no,
+					"bank_code": salary.ifsc_code,
+					"employee_name": salary.employee+": " + salary.employee_name,
+					"currency": frappe.get_cached_value('Company', filters.company, 'default_currency'),
+					"amount": salary.net_pay,
+				}
+
+			data.append(row)
+	return columns, data
+
+def get_account():
+	accounts = frappe.get_all("Account",
+	filters={
+		"account_type": "Bank"
+	}, as_list =1)
+	return accounts
+
+def get_payroll_entry(accounts, filters):
+	accounts = list(itertools.chain(*accounts))
+	payroll_filter = [
+			('payment_account', 'IN', accounts),
+			('number_of_employees', '>', 0),
+			('Company', '=', filters.company)
+		]
+	if filters.to_date:
+		payroll_filter.append(('posting_date', '<', filters.to_date))
+
+	if filters.from_date:
+		payroll_filter.append(('posting_date', '>', filters.from_date))
+
+	entries = get_record("Payroll Entry", payroll_filter, ["name", "payment_account"])
+
+	payment_accounts = [d.payment_account for d in entries]
+	get_company_account(payment_accounts, entries)
+	return entries
+
+def get_salary_slips(payroll_entries):
+	payroll  = [d.name for d in payroll_entries]
+	salary_slips = get_record("Salary Slip", [("payroll_entry", "IN", payroll)],
+		fields = ["modified", "net_pay", "bank_name", "bank_account_no", "payroll_entry", "employee", "employee_name", "status"]
+	)
+
+	payroll_entry_map = {}
+	for entry in payroll_entries:
+		payroll_entry_map[entry.name] = entry
+
+	# appending company debit accounts
+	for slip in salary_slips:
+		slip["debit_acc_no"] = payroll_entry_map[slip.payroll_entry]['company_account']
+
+	return salary_slips
+
+def get_emp_bank_ifsc_code(salary_slips):
+
+	emp_names = [d.employee for d in salary_slips]
+	ifsc_codes = get_record("Employee", [("name", "IN", emp_names)], ["ifsc_code", "name"])
+
+	ifsc_codes_map = {}
+	for code in ifsc_codes:
+		ifsc_codes_map[code.name] = code
+
+	for slip in salary_slips:
+		slip["ifsc_code"] = ifsc_codes_map[code.name]['ifsc_code']
+
+	return salary_slips
+
+def get_company_account(payment_accounts ,payroll_entries):
+	company_accounts = get_record("Bank Account", [("account", "IN", payment_accounts)], ["account", "bank_account_no"])
+	company_accounts_map = {}
+	for acc in company_accounts:
+		company_accounts_map[acc.account] = acc
+
+	for entry in payroll_entries:
+		entry["company_account"] = company_accounts_map[entry.payment_account]['bank_account_no']
+
+	return payroll_entries
+
+def get_record(doctype, filter, fields):
+	return frappe.get_all(doctype, filters=filter, fields=fields)
```

Comment:
Why not use `get_all` directly, this is not needed.

Reply:
done